src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -195,90 +113,129 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
+      if (verificationStatus === "VERIFIED_WITH_CHANGES") {
+        verificationPath = VerificationPath.SUGGESTIONS
+
+        const suggestedOptions = (suggestedAddresses || [])
+          .slice(0, 1)
+          .map(({ address, lines }: any, index) => ({
+            key: `suggestedAddress-${index}`,
+            recommended: index === 0,
+            lines: lines,
+            address: address,
+          }))
+        addressOptions = [...suggestedOptions, inputOption]
       } else {
-        if (verificationStatus === "VERIFIED_WITH_CHANGES") {
-          setModalType(ModalType.SUGGESTIONS)
-          trackEvent({
-            action_type: "validationAddressViewed",
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Confirm your delivery address",
-            option: "suggestions",
-          })
-
-          const suggestedOptions = (suggestedAddresses || [])
-            .slice(0, 1)
-            .map(({ address, lines }: any, index) => ({
-              key: `suggestedAddress-${index}`,
-              recommended: index === 0,
-              lines: lines,
-              address: address,
-            }))
-          setAddressOptions([...suggestedOptions, inputOption])
-        } else {
-          setAddressOptions([inputOption])
-          setModalType(ModalType.REVIEW_AND_CONFIRM)
-          trackEvent({
-            action_type: "validationAddressViewed",
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Check your delivery address",
-            option: "review and confirm",
-          })
-        }
+        addressOptions = [inputOption]
+        verificationPath = VerificationPath.REVIEW_AND_CONFIRM
       }
     }
+  }
+
+  const modalTitle = modalTitles[verificationPath]
+
+  const [selectedAddressKey, setSelectedAddressKey] = useState<
+    AddressOptionKey | undefined
+  >(addressOptions[0]?.key)
+
+  // perform only once on mount
+  useEffect(() => {
+    if (didLoad.current) return
+
+    if (verificationPath === VerificationPath.ERROR_IMMEDIATE_CONFIRM) {
+      const fallbackOption = fallbackFromFormValues(verificationInput)
+      onChosenAddress(AddressVerifiedBy.USER, fallbackOption.address)
+      didLoad.current = true
+    } else if (verificationPath === VerificationPath.IMMEDIATE_CONFIRM) {
+      onChosenAddress(AddressVerifiedBy.ARTSY, addressOptions[0].address!)
+      didLoad.current = true
+    } else {
+      setShowModal(true)
+      didLoad.current = true
+
+      trackViewedModal({
+        subject: modalTitle!,
+      })
+    }
```

Comment:
a previous large useEffect that was running unpredictably has been mostly moved into the imperative outer layer of the code. Instead, we put only the actual side effects (which should only occur once) in here, and make sure it only loads once.

Maybe the `didLoad.current = true` line should just move below L172.

Reply:
Moved down, then removed completely because there was nothing to wait for before loading - firing once on mount (`[]` deps array) does the trick.